### ISSUE TITLE
Cannot connect Dexcom Share to Xdrip - Marshmallow

### ISSUE THREAD

**Issue by Xenocide23** [StephenBlackWasAlreadyTaken/xDrip-Experimental#221]
I have a Oneplus One and recently updated my phone to CM13 and cannot get the dexcom share to sync via bluetooth to any of the Xdrip builds, including the latest experimental branches.  Every time I connect it seems to pair for a few minutes then I get a error 133 or "no bluetooth device to try and connect to".  In addition, while the app seems to think that the share device paired successfully I never get a notification in the actual settings showing a successful pairing completed (ie no smart lock notification).

**Comment by tzachi-dar** [comment]
Do you have location services enabled?
Can you run with logcat enabled? And send the logs.
בתאריך 14 בדצמ 2015 19:52,‏ "Xenocide23" notifications@github.com כתב:

**Comment by Xenocide23** [comment]
Yep, I tried connecting it with all location services enabled, just GPS, and even disabled it to see if the popup requesting location services would come up (it did).  If there is any more information I could provide just let me know.  I'm sure I'm not the only one having this issue and would like to help getting this amazing app working again for everyone.

**Comment by tzachi-dar** [comment]
Well, you need to connect the phone to the usb port of a computer and run
adb logcat and send us the output.
בתאריך 14 בדצמ 2015 20:42,‏ "Xenocide23" notifications@github.com כתב:

**Comment by Xenocide23** [comment]
I have attached a logcat of a fresh install of CM13 with only xdrip installed to keep things simpler. Let me know if you need it in a different format. First time uploading a logcat.

[LogCat.txt](https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/files/62129/LogCat.txt)

**Comment by tzachi-dar** [comment]
Hi,

Looked at your log, but I'm afraid, I can not seem to find anything.
It seems that your phone finds the share device and starts to talk with it.
Not sure if there is any response from the share device.

Hopefully someone else on this list can help you more...

Thanks
Tzachi

**Comment by AdrianLxM** [comment]
Hi @Xenocide23,

the Oneplus One has a CM-android. CM-Android has the "privacyguard" that by default has location services disabled for the app.and will only ask you if you want to allow lhem If xDrip would actually access gps services (what is doesn't).
You have to enable the permission for location services manually in the system settings.
I hope this solves your problem.

Thanks to @LadyViktoria who found this solution.

**Comment by Xenocide23** [comment]
So the solution I found for this issue is to connect to the Dexcom Share device using the normal bluetooth menu first and  then immediately connect to the Dexcom share using the built in Bluetooth scan function in Xdrip.  This wasn't required in Lollipop but seems to be necessary to connect on at least some Marshmallow devices. The permissions were not the issue as I had made sure to enable all of them prior to attempting to pair the Dexcom.

**Comment by krystophv** [comment]
I'm on a Nexus 5 running the latest Marshmallow and was experiencing issues getting the Dexcom Share receiver to connect. I had to go into the xDrip application permission settings (Settings -> Apps -> xDrip -> Permissions) and turn on Location manually before any devices would show up when running the Scan for BT from xDrip.

**Comment by blessedwhitney** [comment]
User Krystophv 's suggestion solved it for me. The beta did not request any permissions upon installation. Going into the settings and manually giving xdrip location permission was needed. From there, worked like a charm.
